src/main/java/com/ericsson/ei/rules/RulesHandler.java

```diff
@@ -39,57 +45,105 @@
 
 @Component
-@Scope(value="thread", proxyMode = ScopedProxyMode.TARGET_CLASS)
+@Scope(value = "thread", proxyMode = ScopedProxyMode.TARGET_CLASS)
 public class RulesHandler {
     private static final Logger LOGGER = LoggerFactory.getLogger(RulesHandler.class);
 
-    @Value("${rules.path}") private String jsonFilePath;
+    private static final String EI_HOME_DEFAULT_NAME = ".eiffel-intelligence";
+
+    @Value("${rules.path}")
+    private String rulesFilePath;
 
     private JmesPathInterface jmesPathInterface = new JmesPathInterface();
     private static String jsonFileContent;
     private static JsonNode parsedJson;
-    
+
     public RulesHandler() {
         super();
     }
-    
+
     public void setParsedJson(String jsonContent) throws JsonProcessingException, IOException {
         ObjectMapper objectmapper = new ObjectMapper();
         parsedJson = objectmapper.readTree(jsonContent);
     }
+
+    public String readRulesFileContent(String path) throws Exception {
+        setRulesFilePath(path);
+        downloadRulesFile(path);
+        return readRulesJsonFileContent();
+    }
     
-    public String readRuleFileContent(String ruleFilePath) throws IOException {
-        String ruleJsonFileContent;
-        InputStream in = this.getClass().getResourceAsStream(ruleFilePath);
-        if(in == null) {
-            ruleJsonFileContent = FileUtils.readFileToString(new File(ruleFilePath), Charset.defaultCharset());
+    public void setRulesFilePath(String path) throws IOException {
+        if (checkIfPathIsURL(path)) {
+            rulesFilePath = getRulesFilePath();
+        } else {
+            rulesFilePath  = path;
+        }
+    }
+
+    public void downloadRulesFile(String path) throws IOException {
+        if (checkIfPathIsURL(path)) {
+            downloadRuleFileToHomeDirectory(path);
+        }
+    }
+
+    public void downloadRuleFileToHomeDirectory(String url) throws IOException {
+        URL source = new URL(url);
+        final File destination = new File(rulesFilePath);
+        FileUtils.copyURLToFile(source, destination);
+    }
+
+    public String readRulesJsonFileContent() throws Exception {
+        String rulesJsonFileContent = null;
+        InputStream inputStream = this.getClass().getResourceAsStream(rulesFilePath);
+        if (inputStream == null) {
+            rulesJsonFileContent = FileUtils.readFileToString(new File(rulesFilePath), Charset.defaultCharset());
         } else {
-            ruleJsonFileContent = getContent(in);
+            rulesJsonFileContent = getContent(inputStream);
+        }
+        if (rulesJsonFileContent.equals("")) {
+            throw new Exception("Rules content cannot be empty");
+        }
+        return rulesJsonFileContent;
+    }
+
+    public boolean checkIfPathIsURL(String path) {
+        String protocolRegex = "https?";
+        try {
+            URI uri = new URI(path);
+            return uri.getScheme().matches(protocolRegex);
```

Comment:
I would also want another check to get file scheme. Let's say the user writes e.g: "file://my-directory/rules.json". This should be interpreted as file, and if it's prefixed with http/https(?) it's a URL. And default value (to keep backwards compatibility) is to assume file path from string if not prefixed.

Reply:
There should be support for this now. I have not had time to test my commit yet though.

Reply:
I've tested successfully with both full file paths and relative file paths, with and without file:/ scheme :) It seems to work, so I'll approve

Reply:
Does it still load internal ExtractionRules with path: "/EiffelAgen...ExtrationRules.json" ?